example/merge_intervals.cpp

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2023 Jiri Nytra (jiri.nytra at gmail dot com)
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <flux.hpp>
+
+#include <cstddef>
+#include <iostream>
+#include <vector>
+
+struct interval_t
+{
+    std::size_t begin;
+    std::size_t end;
+};
+
+std::ostream& operator<<(std::ostream& s, const interval_t& i)
+{
+    return s << '(' << i.begin << ',' << i.end << ')';
+}
+
+bool is_overlapped(interval_t a, interval_t b)
+{
+    return a.end >= b.begin;
+}
+
+auto merge = [](auto seq) -> interval_t
```

Comment:
You could add a constraint to the lambda auto param here, i.e. `[](flux::sequence auto) -> interval_t` to make it clear what kind of parameter the lambda is expecting

Reply:
Good point.